src/main/java/org/eclipse/xpanse/terraform/boot/terraform/TerraformToolHelper.java

```diff
@@ -0,0 +1,162 @@
+import org.eclipse.xpanse.terraform.boot.terraform.utils.SystemCmd;
+import org.eclipse.xpanse.terraform.boot.terraform.utils.SystemCmdResult;
+import org.springframework.stereotype.Component;
+
+/**
+ * Defines methods for handling the terraform tool with version.
+ */
+@Slf4j
+@Component
+public class TerraformToolHelper {
+
+    private static final String OS_NAME = System.getProperty("os.name").toLowerCase();
+    private static final String OS_ARCH = System.getProperty("os.arch").toLowerCase();
+
+    private static final Pattern TERRAFORM_VERSION_PATTERN =
+            Pattern.compile("^(=|>=|<=)([vV])?\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
+    private static final Pattern START_WITH_OPERATOR_PATTERN =
+            Pattern.compile("^(=|>=|<=)");
+    private static final Pattern END_WITH_VERSION_NUMBER_PATTERN =
+            Pattern.compile("^\\D*\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
+    private static final int MAX_VERSION_LENGTH = 20;
+
+    @Resource
+    private SystemCmd systemCmd;
+
+    /**
+     * Get Terraform version number from version string.
+     *
+     * @param version the version string
+     * @return the version number
+     */
+    public String getTerraformVersionNumber(String version) {
+        Matcher matcher = validateVersionFormat(version);
+        return matcher.group(0).replaceAll("^(=|>=|<=)", "");
+    }
+
+    private Matcher validateVersionFormat(String version) {
+        if (StringUtils.isBlank(version)) {
+            throw new InvalidTerraformToolException("Terraform version could not be empty.");
+        }
+        if (version.length() > MAX_VERSION_LENGTH) {
+            throw new InvalidTerraformToolException("Terraform version is too long.");
+        }
+        String noSpaceVersion = version.replaceAll("\\s", "");
+        Matcher matcher = TERRAFORM_VERSION_PATTERN.matcher(noSpaceVersion);
+        if (!matcher.matches()) {
+            StringBuilder errorMsg = new StringBuilder();
+            errorMsg.append(String.format("Invalid terraform version: %s. ", version));
+            if (!START_WITH_OPERATOR_PATTERN.matcher(noSpaceVersion).matches()) {
+                errorMsg.append("Missing or invalid operator (allowed: =, >=, <=). ");
+            }
+            if (!END_WITH_VERSION_NUMBER_PATTERN.matcher(noSpaceVersion).matches()) {
+                errorMsg.append("Missing or invalid version number (format: x.y.z).");
+            }
+            throw new InvalidTerraformToolException(errorMsg.toString());
+        }
+        return matcher;
+    }
+
+
+    /**
+     * Check if the terraform with version is installed.
+     *
+     * @param terraformExecutorFile the terraform executor file
+     * @param versionNumber         the version number
+     * @return true if the terraform with version is installed, otherwise false.
+     */
+    public boolean checkIfTerraformInstalled(File terraformExecutorFile, String versionNumber) {
```

Comment:
does this method even handle if the version number is greater or lesser is available?